hackathon_site/dashboard/frontend/src/components/inventory/InventoryFilter/InventoryFilter.js

```diff
@@ -1,105 +1,188 @@
 import React from "react";
 import styles from "./InventoryFilter.module.scss";
+import { Formik, Field } from "formik";
+
 import Typography from "@material-ui/core/Typography";
 import Button from "@material-ui/core/Button";
 import Chip from "@material-ui/core/Chip";
 import Divider from "@material-ui/core/Divider";
 import Checkbox from "@material-ui/core/Checkbox";
 import Radio from "@material-ui/core/Radio";
 import RadioGroup from "@material-ui/core/RadioGroup";
+import FormGroup from "@material-ui/core/FormGroup";
 import FormControlLabel from "@material-ui/core/FormControlLabel";
+import { inventoryCategories } from "testing/mockData";
+import CircularProgress from "@material-ui/core/CircularProgress";
+
+const RadioOrderBy = ({ field, options }) => (
+    <RadioGroup {...field} name={field.name}>
+        {options.map((item, i) => (
+            <FormControlLabel
+                name={field.name}
+                value={item.value}
+                label={item.label}
+                control={<Radio color="primary" />}
+                checked={field.value === item.value}
+                key={i}
+            />
+        ))}
+    </RadioGroup>
+);
 
-const InventoryFilter = ({ categories, applyFilter, removeFilter }) => (
+const CheckboxCategory = ({ field, options }) => (
+    <FormGroup {...field} name={field.name}>
+        {options.map((item, i) => (
+            <div className={styles.filterCategory} key={i}>
+                <FormControlLabel
+                    name={field.name}
+                    value={item.name}
```

Comment:
We should be using the category ID here as the value, since that's what the API takes. I think this works:

```javascript
<FormControlLabel
    name={field.name}
    value={item.id}
    control={<Checkbox color="primary" />}
    label={item.name}
    checked={field.value.includes(item.id.toString())}
/>
```

Then in mockData.js:

```javascript
export const inventoryCategories = [
    { id: 1, name: "MCU", qty: 12 },
    { id: 2, name: "MCU_limit_3", qty: 6 },
    { id: 3, name: "FPGA", qty: 9 },
    { id: 4, name: "Sensors", qty: 21 },
    { id: 5, name: "Sensors_limit_2", qty: 3 },
    { id: 6, name: "Peripherals", qty: 389 },
    { id: 7, name: "1080p_cameras", qty: 4 },
    { id: 8, name: "Grove", qty: 22 },
    { id: 9, name: "Grove_style_boards", qty: 12 },
    { id: 10, name: "Grove_style_modules", qty: 10 },
];
```

Only problem with this is that the ID values become strings in the form, which will require massaging to ints for the API. Is there any way around that?

Reply:
Didn't find out how to do it without parsing all the ids before submitting the form values